consensus/ethash/algorithm_test.go

```diff
@@ -698,6 +698,8 @@ func TestHashimoto(t *testing.T) {
 // Tests that caches generated on disk may be done concurrently.
 func TestConcurrentDiskCacheGeneration(t *testing.T) {
 	// Create a temp folder to generate the caches into
+	// TODO: t.TempDir fails to remove the directory on Windows
```

Comment:
Do you know why is this a problem here but not in the other tests?

Reply:
This happens in `algorithm_test.go: TestConcurrentDiskCacheGeneration` and `ethash_test.go: TestCacheFileEvict` only, where we pass the temporary directory to `Config { CacheDir: tmpdir }`. 

Unfortunately, I don't have a Windows machine so I am unable to further debug this. It _might_ be a Windows specific issue, see https://github.com/golang/go/issues/50051